src/app/shared/components/metrics/metrics.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { VideoItem } from 'src/app/youtube/models/search-item.model';
+import {
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting,
+} from '@angular/platform-browser-dynamic/testing';
+import { MetricsComponent } from './metrics.component';
+
+TestBed.initTestEnvironment(
+    BrowserDynamicTestingModule,
+    platformBrowserDynamicTesting()
+);
+
+describe('MetricsComponent', () => {
+    let component: MetricsComponent;
+    let fixture: ComponentFixture<MetricsComponent>;
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(MetricsComponent);
+        component = fixture.componentInstance;
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should set metricsData based on input item', () => {
+        const mockItem: VideoItem = {
+            id: '123',
+            statistics: {
+                viewCount: '100',
+                likeCount: '50',
+                favoriteCount: '10',
+                commentCount: '30',
+            },
+            snippet: {
+                publishedAt: 'string',
+                title: 'string',
+                description: 'string',
+                thumbnails: {
+                    default: {
+                        url: 'formValue.imageLink',
+                        width: 120,
+                        height: 90,
+                    },
+                    medium: {
+                        url: 'formValue.imageLink',
+                        width: 320,
+                        height: 180,
+                    },
+                    high: {
```

Comment:
just a note, that a good practice is to have a file with such mock that you can reuse for tests in the whole project